packages/hdwallet-provider/src/index.ts

```diff
@@ -168,9 +168,10 @@ class HDWalletProvider {
       ).toLowerCase();
 
       switch (providerProtocol) {
-        case "ws":
-        case "wss":
+        case "ws:":
+        case "wss:":
```

Comment:
Huh. That's weird. Maybe the better solution would be to strip off the `:` before the `switch` statement.

But I don't care enough on this point to block approval. We'll see what @eggplantzzz says for the second approve :)

Thanks again or this @roderik!

Reply:
Yeah the same : are already used here: https://github.com/trufflesuite/truffle/blob/develop/packages/hdwallet-provider/src/index.ts#L215 so i thought just to go with the flow (even though obviously nobody uses this undocumented feature :))

Reply:
sounds good!